backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/dialog/BpkDialog.kt

```diff
@@ -0,0 +1,158 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.dialog
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.window.DialogProperties
+import net.skyscanner.backpack.compose.dialog.internal.BpkDialogImpl
+import net.skyscanner.backpack.compose.dialog.internal.DialogType
+
+object BpkDialog {
```

Comment:
i wanted to originally make this a sealed class or interface, but in the end decided against it and make the type definition internal and move it to the internal class to make the public vs internal api distinction clear